Suggest: '(& f)' -> 'f'

mlang opened the pull request:
Thanks for the pull request!

By raising this pull request you confirm you are licensing your contribution under all licenses that apply to this project (see LICENSE) and that you have no patents covering your contribution.

If you care, my PR preferences are at https://github.com/ndmitchell/neil#contributions, but they're all guidelines, and I'm not too fussy - you don't have to read them.

ndmitchell reviewed (commented):
Thanks for the hint! I thought we already had this one included in the standard defaults with https://github.com/ndmitchell/hlint/blob/c00bb3d72e88614463983bcbf1b24e5db38e4cb0/data/hlint.yaml#L322 - is that not firing for you for some reason?

mlang commented:
Mind you, I am totally new to hlint internals.  But indeed, the existing
rule for Data.Function.& (which I missed) does not fire for my test
case:

```
***@***.***:~/Projects/Haskell/hlint$ cat test.hs
module Main where

import Control.Lens

main :: IO ()
main = do
  print $ (& (+ 1)) 0

***@***.***:~/Projects/Haskell/hlint$ git checkout master
Switched to branch 'master'
Your branch is up to date with 'upstream/master'.
***@***.***:~/Projects/Haskell/hlint$ hlint --hint data/hlint.yaml test.hs
No hints
***@***.***:~/Projects/Haskell/hlint$ git checkout redunant
Switched to branch 'redunant'
Your branch is up to date with 'origin/redunant'.
***@***.***:~/Projects/Haskell/hlint$ hlint --hint data/hlint.yaml test.hs
test.hs:7:11-19: Warning: Redundant &
Found:
  (& (+ 1))
  Perhaps:
    (+ 1)

1 hint
```

-- 
CYa,
  ⡍⠁⠗⠊⠕

ndmitchell commented:
I've removed the `Data.Function` qualification. At the beginning `&` was a bit of a rare function, so I wanted to be careful about when it triggered. Now it's much more common, so I think relaxing it makes a lot more sense. Thanks for the suggestion!
